Source/gui/SNTBinaryMessageWindowController.m

```diff
@@ -135,13 +150,96 @@ - (IBAction)showCertInfo:(id)sender {
+                          ruleType = SNTRuleTypeBinary;
+                          ruleIdentifier = self.event.fileSHA256;
+                        }
+
+                        // Add rule to allow binary same as santactl rule.
+                        SNTRule *newRule =
+                          [[SNTRule alloc] initWithIdentifier:ruleIdentifier
+                                                        state:SNTRuleStateAllow
+                                                         type:ruleType
+                                                    customMsg:@"Approved by user"
+                                                    timestamp:[[NSDate now] timeIntervalSince1970]];
+
+                        // TODO add bundle hash to a dictionary / database
+                        // table for auto-approval so the user isn't drowning
+                        // in dialogs
+                        dispatch_semaphore_t sema2 = dispatch_semaphore_create(0);
+
+                        MOLXPCConnection *daemonConn =
+                          [SNTXPCControlInterface configuredConnection];
+                        daemonConn.invalidationHandler = ^{
+                          LOGE(@"Connection invalidated");
+                          dispatch_semaphore_signal(sema2);
+                        };
+
+                        [daemonConn resume];
+                        [[daemonConn remoteObjectProxy]
+                          addStandaloneRule:newRule
+                                      reply:^(NSError *error) {
+                                        if (error) {
+                                          LOGE(@"Failed to modify rules standalone: %s",
+                                               [error.localizedDescription UTF8String]);
+                                          LOGE(@"Failure reason: %@", error.localizedFailureReason);
+                                        }
+                                        dispatch_semaphore_signal(sema2);
+                                      }];
+                        dispatch_semaphore_wait(sema2, DISPATCH_TIME_FOREVER);
+                      } else {
+                        LOGE(@"Authentication Error: %@", error);
+                      }
+                      dispatch_semaphore_signal(sema);
+                    }];
+  dispatch_semaphore_wait(sema, DISPATCH_TIME_FOREVER);
+}
+
 - (IBAction)openEventDetails:(id)sender {
+  if ([[SNTConfigurator configurator] enableStandaloneMode]) {
+    [self closeWindow:sender];
+    [self approveBinaryForStandaloneMode];
+    return;
+  }
+
   NSURL *url = [SNTBlockMessage eventDetailURLForEvent:self.event customURL:self.customURL];
 
   [self closeWindow:sender];
   [[NSWorkspace sharedWorkspace] openURL:url];
 }
 
+- (IBAction)checkEventDetails:(id)sender {
+  NSString *eventCheckURL =
+    [NSString stringWithFormat:@"https://www.virustotal.com/gui/search/%@", self.event.fileSHA256];
```

Comment:
I like this idea. Though we should make it configurable. Similar to what we do for event details.

I don't think we want to hard code virus total here.